addons/sale/models/sale_order.py

```diff
@@ -1717,11 +1717,12 @@ def _track_finalize(self):
         return super()._track_finalize()
 
     @api.returns('mail.message', lambda value: value.id)
-    def message_post(self, **kwargs):
-        if self.env.context.get('mark_so_as_sent'):
+    def message_post(self, mark_so_as_sent=False, **kwargs):
```

Comment:
Don't think we need to change message_post API. Would instead check to override ``_post_message`` on schedule message model, and take back the parameter from notification_parameters, update context, then call super() to propagate it indirectly to message_post, something like that.